pkg/instance/service.go

```diff
@@ -94,29 +100,89 @@ func (s service) validateDeployment(deployment *model.Deployment) error {
 		return err
 	}
 
-	stacks, err := s.collectStacks(deployment)
-	if err != nil {
-		return err
-	}
-
-	err = stack.ValidateConsumedParameters(stacks)
+	err = s.validateDeploymentParameters(deployment)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (s service) collectStacks(deployment *model.Deployment) ([]model.Stack, error) {
-	stacks := make([]model.Stack, len(deployment.Instances))
-	for i, instance := range deployment.Instances {
+func (s service) validateDeploymentParameters(deployment *model.Deployment) error {
```

Comment:
`validateDeploymentParameters` is and should be called after `resolveParameters`.

I think the validation can be simpler 🤔. The rule is every stack parameter needs to have a value. Since you already resolved default values there is no need to add any logic regarding that. There is also no need to go through graph of where a parameter is consumed from for the same reason.

The only question is if a parameter does not have a value: who's fault is it?
* if the parameter is a consumed one it is likely our fault
* if the parameter is not a consumed one it is likely the users fault

So these two cases need to influence the error message. Unless I miss something.

I would also call `validateDeploymentParameters` on deploy. This way we can prevent some calls to `helmfile` which would fail due to some changes in the stacks.

I am unsure if we should call `resolveParameters` again on deploy. We already did that on save and stored the parameters right? 🤔